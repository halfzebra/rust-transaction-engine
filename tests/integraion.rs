#[cfg(test)]
mod integraion {
    use assert_cmd::prelude::*;
    use std::process::Command;
    use std::str;

    #[test]
    fn basic() -> Result<(), Box<dyn std::error::Error>> {
        let mut cmd = Command::cargo_bin("rust-transaction-engine")?;
        cmd.arg("./examples/basic.csv");
        let stdout = String::from_utf8(cmd.assert().success().get_output().stdout.clone())?;

        insta::assert_snapshot!(stdout);

        Ok(())
    }

    #[test]
    fn same_tx() -> Result<(), Box<dyn std::error::Error>> {
        let mut cmd = Command::cargo_bin("rust-transaction-engine")?;
        cmd.arg("./examples/same-tx.csv");
        let stdout = String::from_utf8(cmd.assert().success().get_output().stdout.clone())?;

        insta::assert_snapshot!(stdout);

        Ok(())
    }
}
