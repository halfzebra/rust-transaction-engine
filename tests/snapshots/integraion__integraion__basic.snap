---
source: tests/integraion.rs
expression: "String::from_utf8(cmd.assert().success().get_output().stdout.clone())?"
---
client,available,held,total,locked
1,1.5,0.0,1.5,false
2,2.0,0.0,2.0,false

